spec/models/member_spec.rb

```diff
@@ -160,4 +163,165 @@
       it { is_expected.not_to be_thredded_admin }
     end
   end
+
+	describe "renewed_subscription" do
+
+			before(:each) do
+				@member = FactoryBot.create(:member)
+			end
+
+		context "when is the first subscription" do
+
+			it { 
+				allow(Date).to receive(:today) { Date.new(2001, 2, 5) }
+				@member.renewed_subscription
+				expect(@member.end_subscription).to eq(Date.new(2002, 2, 4)) 
+			}
+
+			it { 
+				allow(Date).to receive(:today) { Date.new(2121, 7,21) }
+				@member.renewed_subscription
+				expect(@member.end_subscription).to eq(Date.new(2122, 7, 20)) 
+			}
+
+			it { 
+				allow(Date).to receive(:today) { Date.new(2023, 2, 1) }
+				@member.renewed_subscription
+				expect(@member.end_subscription).to eq(Date.new(2024, 1, 31)) 
+			}
+
+			it { 
+				allow(Date).to receive(:today) { Date.new(2100, 11, 15) }
+				@member.renewed_subscription
+				expect(@member.end_subscription).to eq(Date.new(2101, 11, 14)) 
+			}
+
+			it { 
+				allow(Date).to receive(:today) { Date.new(2020, 2, 5) }
+				@member.renewed_subscription
+				expect(@member.end_subscription).to eq(Date.new(2021, 2, 4)) 
+			}
+
+			it { 
+				allow(Date).to receive(:today) { Date.new(2399, 10, 24) }
+				@member.renewed_subscription
+				expect(@member.end_subscription).to eq(Date.new(2400, 10, 23)) 
+			}
+
+			it { 
+				allow(Date).to receive(:today) { Date.new(2060, 2, 29) }
+				@member.renewed_subscription
+				expect(@member.end_subscription).to eq(Date.new(2061, 2, 28)) 
+			}
```

Comment:
You are testing the same single case multiple times : checking that the new subscription date will equal `today + 1 year`
If you want to test random values, you can set `today` as a random date.